packages/input-option/src/InputOption/InputOption.tsx

```diff
@@ -3,57 +3,78 @@ import React from 'react';
 import { cx } from '@leafygreen-ui/emotion';
 import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
 import {
-  Polymorphic,
-  PolymorphicAs,
-  usePolymorphic,
+  InferredPolymorphic,
+  useInferredPolymorphic,
 } from '@leafygreen-ui/polymorphic';
 
 import {
-  inputOptionActiveStyles,
-  inputOptionDisabledStyles,
-  inputOptionHoverStyles,
+  disabledStyles,
+  getFormElementStyle,
+  getMenuElementStyle,
   inputOptionStyles,
-  inputOptionThemeStyles,
   inputOptionWedge,
-  titleSelectionStyles,
 } from './InputOption.style';
-import { InputOptionProps } from './InputOption.types';
+import {
+  ActionType,
+  InputOptionProps,
+  RenderedContext,
+} from './InputOption.types';
 
-export const InputOption = Polymorphic<InputOptionProps>(
+export const InputOption = InferredPolymorphic<InputOptionProps, 'div'>(
   (
     {
-      as = 'li' as PolymorphicAs,
+      as,
       children,
       disabled,
       highlighted,
-      selected,
+      checked,
       darkMode: darkModeProp,
       showWedge = true,
       isInteractive = true,
       className,
+      actionType = ActionType.Default,
+      renderedContext = RenderedContext.FormElement,
       ...rest
     },
     ref,
   ) => {
-    const { Component } = usePolymorphic(as);
+    const { Component } = useInferredPolymorphic(as, rest, 'div');
     const { theme } = useDarkMode(darkModeProp);
+
+    const themedStatefulStyles =
+      renderedContext === RenderedContext.FormElement
+        ? getFormElementStyle({
+            theme,
+            checked,
+            highlighted,
+            disabled,
+            showWedge,
+            isInteractive,
+          })
+        : getMenuElementStyle({
+            theme,
+            checked,
+            highlighted,
+            disabled,
+            showWedge,
+            actionType,
+            isInteractive,
+          });
```

Comment:
Feels weird to me there's some styling logic here on the component itself when we have a function dedicated to styling logic in the styles.ts file. I think, if possible, we should use `cx` to split styles on the component and `css` strings to define the styles. If there's a need to define more complex logic, we can use nested style objects using Record<Theme, Record<SomeProp, string>> or inline ternaries inside a simple `() => css` function

Reply:
it's just deciding which element styles to use - i can wrap those two functions in another function if you prefer?

Reply:
I agree, this feels off. Seems like `renderedContext` should just be an additional prop passed into a single function